pandas/tests/indexes/interval/test_interval_tree.py

```diff
@@ -13,14 +13,27 @@ def dtype(request):
     return request.param
 
 
-@pytest.fixture(scope='class')
-def tree(dtype):
-    left = np.arange(5, dtype=dtype)
+@pytest.fixture(scope='class', params=[
+    np.arange(5, dtype='int64'),
+    np.arange(5, dtype='int32'),
+    np.arange(5, dtype='uint64'),
+    np.arange(5, dtype='float64'),
+    np.arange(5, dtype='float32'),
+    np.array([0, 1, 2, 3, 4, np.nan], dtype='float64'),
+    np.array([0, 1, 2, 3, 4, np.nan], dtype='float32')])
+def tree(request):
+    left = request.param
     return IntervalTree(left, left + 2)
 
 
 class TestIntervalTree(object):
 
+    def test_construction_nan(self):
+        # GH 23352
+        left, right = [0, 1, 2, np.nan], [1, 2, 3, np.nan]
+        with tm.assert_produces_warning(None):
+            IntervalTree(left, right, leaf_size=2)
```

Comment:
what exactly does this produce though? are the nan's valid? you are just dropping them.

Reply:
The intent here was to verify that the `RuntimeWarning` described in the issue is not occurring.  I guess this test isn't really necessary, as at least one CI instance is set to fail if warnings are raised.  Removed this test, and instead parametrized the `tree` fixture to include `leaf_size` that will trigger the warning and cause the invalid behavior described in the issue (so beyond just a warning would cause the asserts to fail if there's a regression).